src/v/cluster_link/source_topic_syncer.cc

```diff
@@ -117,6 +120,271 @@ ss::future<> source_topic_syncer::run_impl() {
+                  cache_entry.replication_factor);
+            }
+            continue;
+        }
+        for (const auto& c : config_resp_it->configs) {
+            if (c.value.has_value()) {
+                vlog(
+                  logger().trace,
+                  "Config {} for topic {}: {}",
+                  c.name,
+                  topic,
+                  c.value.value());
+                auto metadata_it = metadata.topic_configs.find(c.name);
+                if (
+                  metadata_it == metadata.topic_configs.end()
+                  || metadata_it->second != c.value.value()) {
+                    vlog(
+                      logger().trace,
+                      "Config {} has changed for topic {}",
+                      c.name,
+                      topic);
+                    enqueue_command = true;
+                }
+            }
+        }
+
+        if (enqueue_command) {
+            chunked_hash_map<ss::sstring, ss::sstring> topic_configs;
+            topic_configs.reserve(config_resp_it->configs.size());
+            for (const auto& conf : config_resp_it->configs) {
+                if (conf.value.has_value()) {
+                    topic_configs.emplace(conf.name, conf.value.value());
+                }
+            }
+            model::update_mirror_topic_properties_cmd cmd{
+              .topic = topic,
+              .partition_count = static_cast<int32_t>(
+                topic_cache_it->second.partitions.size()),
+              .replication_factor = topic_cache_it->second.replication_factor,
+              .topic_configs = std::move(topic_configs)};
+            vlog(logger().trace, "Enqueuing update command: {}", cmd);
+            update_properties_cmds.emplace_back(std::move(cmd));
+        }
+    }
+
+    // Now that we have built up the list of commands, we can now execute them,
+    // first start by failing the mirror topics
+    for (const auto& t : topics_to_fail) {
+        vlog(logger().warn, "Failing mirror topic {}", t);
+        auto res = co_await get_link()->update_mirror_topic_state(
+          model::update_mirror_topic_state_cmd{
+            .topic = t, .state = model::mirror_topic_state::failed});
+        if (res != ::cluster::cluster_link::errc::success) {
+            vlog(logger().error, "Failed to fail mirror topic {}: {}", t, res);
+        } else {
+            vlog(logger().info, "Successfully failed mirror topic {}", t);
+        }
+    }
+
+    // Now update the properties of the mirror topics
+    for (auto& cmd : update_properties_cmds) {
+        vlog(logger().trace, "Updating mirror topic properties: {}", cmd);
+        auto res = co_await get_link()->update_mirror_topic_properties(
+          std::move(cmd));
+        if (res != ::cluster::cluster_link::errc::success) {
+            vlog(
+              logger().error,
+              "Failed to update mirror topic properties for {}: {}",
+              cmd.topic,
+              res);
```

Comment:
does this have any consequences with respect to the call site or the cluster linking system at large? or this is all best-effort?